server/src/main/java/org/cloudfoundry/identity/uaa/scim/event/GroupModifiedEvent.java

```diff
@@ -145,47 +141,4 @@ public String toString() {
                 '}';
         }
     }
-
-    protected static Authentication getContextAuthentication() {
-        Authentication a = SecurityContextHolder.getContext().getAuthentication();
-        if (a==null) {
-            a = new Authentication() {
-                ArrayList<GrantedAuthority> authorities = new ArrayList<GrantedAuthority>();
-                @Override
-                public Collection<? extends GrantedAuthority> getAuthorities() {
-                    return authorities;
-                }
-
-                @Override
-                public Object getCredentials() {
-                    return null;
-                }
-
-                @Override
-                public Object getDetails() {
-                    return null;
-                }
-
-                @Override
-                public Object getPrincipal() {
-                    return "null";
-                }
-
-                @Override
-                public boolean isAuthenticated() {
-                    return false;
-                }
-
-                @Override
-                public void setAuthenticated(boolean isAuthenticated) throws IllegalArgumentException {
-                }
-
-                @Override
-                public String getName() {
-                    return "null";
-                }
-            };
-        }
-        return a;
-    }
```

Comment:
This block of code was just removed. I supposed that means it was not being used? Or was this the duplicate code?

Reply:
yes because sonar shows the duplication and I simply remove this and removed one part and let the other run in both cases, see https://sonarcloud.io/code?id=cloudfoundry-identity-parent&selected=cloudfoundry-identity-parent%3Aserver%2Fsrc%2Fmain%2Fjava%2Forg%2Fcloudfoundry%2Fidentity%2Fuaa%2Faudit%2Fevent%2FAbstractUaaEvent.java

or see here how sonars makes it visible

![image](https://github.com/cloudfoundry/uaa/assets/11627201/76de2bcb-b466-434e-a1c4-81e897060435)